crates/primitives/src/botanix/mint_validation.rs

```diff
@@ -207,6 +208,13 @@ pub fn try_parse_mint_event(
                     error!("Failed to parse pegin meta: {:?}", err);
                     err
                 })?;
+            if !outpoints.insert(proof.outpoint().clone()) {
+                return Err(ParseMintEventError::InvalidPeginData {
+                    error: PeginDataError::Invalid("Duplicate outpoint hence invalid pegin meta"),
```

Comment:
Just an error description would do. The internal variants shouldnt justify why they exists. 

```suggestion
                    error: PeginDataError::Invalid("Duplicate outpoint"),
```

Reply:
Done!